x-pack/plugins/security_solution/public/resolver/view/limit_warnings.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiCallOut } from '@elastic/eui';
+import { FormattedMessage } from 'react-intl';
+
+const LineageLimitMessage = (
+  <>
+    <FormattedMessage
+      id="xpack.securitySolution.endpoint.resolver.eitherLineageLimitExceeded"
+      defaultMessage="Some process events in the visualization and event list below could not be displayed because the data limit has been reached."
+    />
+  </>
+);
+
+const LineageTitleMessage = React.memo(function RelatedEventsLimitMessage({
+  numberOfEntries,
+}: {
+  numberOfEntries: number;
+}) {
+  return (
+    <>
+      <FormattedMessage
+        id="xpack.securitySolution.endpoint.resolver.relatedEventLimitTitle"
+        defaultMessage="This list includes {numberOfEntries} process events."
+        values={{ numberOfEntries }}
+      />
+    </>
+  );
+});
+
+const RelatedEventsLimitMessage = React.memo(function RelatedEventsLimitMessage({
+  category,
+  numberOfEventsMissing,
+}: {
+  numberOfEventsMissing: number;
+  category: string;
+}) {
+  return (
+    <>
+      <FormattedMessage
+        id="xpack.securitySolution.endpoint.resolver.relatedEventLimitExceeded"
+        defaultMessage="{numberOfEventsMissing} {category} events could not be displayed because the data limit has been reached."
+        values={{ numberOfEventsMissing, category }}
+      />
+    </>
+  );
+});
+
+const RelatedLimitTitleMessage = React.memo(function RelatedLimitTitleMessage({
+  category,
+  numberOfEventsDisplayed,
+}: {
+  numberOfEventsDisplayed: number;
+  category: string;
+}) {
+  return (
+    <>
+      <FormattedMessage
+        id="xpack.securitySolution.endpoint.resolver.relatedLimitsExceededTitle"
+        defaultMessage="This list includes {numberOfEventsDisplayed} {category} events."
+        values={{ numberOfEventsDisplayed, category }}
+      />
+    </>
+  );
+});
+
+/**
+ * Limit warning for hitting the /events API limit
+ */
+export const RelatedEventLimitWarning = React.memo(function RelatedEventLimitWarning({

```

Comment:
Maybe instead of 
```
  eventType: string;
  matchingEventEntries: unknown[];
  aggregateCountForEventType: number;
```

this could instead take:

```
  eventType: string;
  numberActuallyDisplayed: number;
  numberMissing: number;
```

This would allow you to move the business logic out of this component.

1. component now has a single responsibility: visual formatting of those numbers
2. the business logic can be centralized in the selectors and unit tested there.